lib/feature/wallet/view/wallet_page_wm.dart

```diff
@@ -99,40 +103,36 @@ class WalletPageWidgetModel
     );
   }
 
-  void _onAccountChanged() {
-    _checkBadge();
-    _checkUnconfirmedTransactions();
+  void _onAccountChanged(KeyAccount account) {
+    _checkBadge(account);
+    _checkUnconfirmedTransactions(account);
   }
 
-  void _checkBadge() {
-    //check user create new wallet or login
-    final account = _currentAccount.value;
+  void _checkBadge(KeyAccount account) {
     final isNewUser = model.isNewUser();
 
-    if (account == null) return;
-
-    if (isNewUser != null) {
-      if (isNewUser) {
-        _isShowingNewTokensNotifier.accept(true);
-        _isShowingBadgeNotifier.accept(true);
-      } else {
-        _isShowingBadgeNotifier.accept(false);
-        model.hideShowingBadge(account);
-      }
-
-      model.resetValueNewUser();
+    if (isNewUser == null) {
+      _isShowingBadgeNotifier.accept(model.isShowingBadge(account) ?? true);
+      _isShowingNewTokensNotifier.accept(
+        model.isShowingNewTokens(account) ?? true,
+      );
       return;
     }
 
-    _isShowingBadgeNotifier.accept(model.isShowingBadge(account) ?? true);
-    _isShowingNewTokensNotifier.accept(
-      model.isShowingNewTokens(account) ?? true,
-    );
+    if (isNewUser) {
+      _isShowingNewTokensNotifier.accept(true);
+      _isShowingBadgeNotifier.accept(true);
+    } else {
+      _isShowingNewTokensNotifier.accept(true);
+      _isShowingBadgeNotifier.accept(false);
+      model.hideShowingBadge(account);
+    }
```

Comment:
```suggestion
    _isShowingBadgeNotifier.accept(isNewUser);
    _isShowingNewTokensNotifier.accept(true);

    if (!isNewUser) {
      model.hideShowingBadge(account);
    }
```
По идее можно чуть упростить дублирующиеся ветки: